tests/unit-tests/NestedFieldSupportForArrayData.swift

```diff
@@ -0,0 +1,134 @@
+                                               "dataType": "user",
+                                               "field": "furniture.furnitureType",
+                                               "comparatorType": "Equals",
+                                               "value": "Sofa",
+                                               "fieldType": "string"
+                                           }
+                                       ]
+                                   }
+                               }
+                           ]
+                       }
+                   ]
+               }
+           }
+         ]
+       }
+    """
+
+    override func setUp() {
+        super.setUp()
+    }
+
+    func data(from jsonString: String) -> Data? {
+        return jsonString.data(using: .utf8)
+    }
+
+    func testNestedFieldSuccess() {
+
+        let eventItems: [[AnyHashable: Any]] = [
+            [
+                "dataType":"user",
+                 "email":"user@example.com",
+                 "dataFields":[
+                    "furniture": [
+                        [
+                            "furnitureType": "Sofa",
+                            "furnitureColor": "White",
+                            "lengthInches": 40,
+                            "widthInches": 60
+                        ],
+                        [
+                            "furnitureType": "Sofa",
+                            "furnitureColor": "Gray",
+                            "lengthInches": 20,
+                            "widthInches": 30
+                        ],
+                    ]
+                ]
+            ]
+        ]
+
+
+        let expectedCriteriaId = "168"
+        let matchedCriteriaId = CriteriaCompletionChecker(anonymousCriteria: data(from: mockData)!, anonymousEvents: eventItems).getMatchedCriteria()
+        XCTAssertEqual(matchedCriteriaId, expectedCriteriaId)
+    }
+
+    func testNestedFieldFailed() {
+
+        let eventItems: [[AnyHashable: Any]] = [
+            [
+                "dataType":"user",
+                 "email":"user@example.com",
+                 "dataFields":[
+                    "furniture": [
+                        [
+                            "furnitureType": "Sofa",
+                            "furnitureColor": "White",
+                            "lengthInches": 40,
+                            "widthInches": 60
+                        ],
+                        [
+                            "furnitureType": "Table",
+                            "furnitureColor": "Gray",
+                            "lengthInches": 20,
+                            "widthInches": 30
+                        ],
+                    ]
+                ]
+            ]
+        ]
+
+        let matchedCriteriaId = CriteriaCompletionChecker(anonymousCriteria: data(from: mockData)!, anonymousEvents: eventItems).getMatchedCriteria()
+        XCTAssertEqual(matchedCriteriaId, nil)
```

Comment:
Same comment here as I left for Android.

Reply:
Updated the logic as per the PR comment